pkgs/applications/networking/cluster/kubebuilder/default.nix

```diff
@@ -0,0 +1,50 @@
+{ lib
+, buildGoModule
+, fetchFromGitHub
+, installShellFiles
+, makeWrapper
+, git
+, go ? false }:
+
+buildGoModule rec {
+  pname = "kubebuilder";
+  version = "3.1.0";
+
+  src = fetchFromGitHub {
+    owner = "kubernetes-sigs";
+    repo = "kubebuilder";
+    url = "https://github.com/kubernetes-sigs/kubebuilder.git";
+    rev = "v${version}";
+    leaveDotGit = true;
+    sha256 = "1726j2b5jyvllvnk60g6px3g2jyyphd9pc4vgid45mis9b60sh8a";
+  };
+  vendorSha256 = "0zxyd950ksjswja64rfri5v2yaalfg6qmq8215ildgrcavl9974n";
+
+  subPackages = ["cmd" "pkg/..."];
+
+  preBuild = ''
+    export buildFlagsArray+=("-ldflags=-X main.kubeBuilderVersion=v${version} \
+        -X main.goos=$GOOS \
+        -X main.goarch=$GOARCH \
+        -X main.gitCommit=$(git rev-parse HEAD) \
+        -X main.buildDate=$(git log -1 --date=iso-strict --format=%cd)")
```

Comment:
Can we set those to some variant of version to not need to fetch the .git directory?

Reply:
Hmm, I could see setting `gitCommit` to the tag (`v${version}`) since that basically references the commit. If I don't set buildDate though, it will default to the UNIX epoch:

```
[nix-shell:~/Projects/nixpkgs]$ kubebuilder version
Version: main.version{KubeBuilderVersion:"v3.1.0", KubernetesVendor:"unknown", GitCommit:"$Format:%H$", BuildDate:"1970-01-01T00:00:00Z", GoOs:"linux", GoArch:"amd64"}
```

I didn't want to set the buildDate to "now" in the build, because that seems like it'd break reproducibility.

I could see the commit hash and timestamp of that commit hash (as a proxy for actual build time) as being more useful to a kubebuilder developer receiving a bug report from someone who installed this package. But, I'm also new here, this is my first Nix package. Is it bad to rely on the .git directory in Nix builds? Is it possible to do a shallow fetch (--depth=1, etc) to minimize the overhead? What would you recommend?